### PR TITLE
dom: DocumentHTML getters

### PR DESCRIPTION
**Pull request by krichprollsch**
relates with #19

### CODE REVIEW

File: src/html/document.zig [hunk 1]
```diff
@@ -18,9 +23,175 @@ pub const HTMLDocument = struct {
...
+            next = try walker.get_next(root, next) orelse return null;
+            if (std.ascii.eqlIgnoreCase("head", try parser.nodeName(next.?))) {
+                return @as(*parser.Head, @ptrCast(next.?));
+            }
+        }
+    }
+
+    // TODO: not implemented by libdom
+    pub fn get_cookie(_: *parser.DocumentHTML) ![]const u8 {
+        return error.NotImplemented;
```
Comment (francisbouvier) [inline src/html/document.zig:65]: Which difference are you making between NotSupported and NotImplemented?

Is there a reason to use error vs parser.DOMError?

File: src/html/document.zig [hunk 2]
```diff
@@ -18,9 +23,175 @@ pub const HTMLDocument = struct {
...
+        return v;
+    }
+
+    pub fn _getElementsByName(self: *parser.DocumentHTML, alloc: std.mem.Allocator, name: []const u8) !NodeList {
+        var list = try NodeList.init();
+        errdefer list.deinit(alloc);
+
+        if (name.len == 0) return list;
+
+        const root = try rootNode(self) orelse return list;
```
Comment (francisbouvier) [inline src/html/document.zig]:
```suggestion
        const root = try self.rootNode() orelse return list;
```

I think it's shows better that `rootNode` is a method of HTMLDocument

File: src/html/document.zig [hunk 3]
```diff
@@ -18,9 +23,175 @@ pub const HTMLDocument = struct {
...
+        while (i < ln) {
+            const n = try c.item(i) orelse break;
+            try list.append(alloc, n);
+            i += 1;
+        }
+
+        return list;
+    }
+
+    inline fn rootNode(self: *parser.DocumentHTML) !?*parser.Node {
```
Comment (francisbouvier) [inline src/html/document.zig]: Just for my understanding, this is mandatory right ? We can't ptrCast directly the DocumentHTML to Node for that?

File: src/html/document.zig (hunk 1 above)
Comment (krichprollsch) [reply src/html/document.zig:65]: My bad, I wanted to use an `error.NotImplemented` everywhere we have  a kind of `TODO`.

File: src/html/document.zig (hunk 2 above)
Comment (krichprollsch) [reply src/html/document.zig]: AFAIK I can't do that b/c `rootNode` is defined into `HTMLDocument` and `self` is a `*parser.DocumentHTML`

File: src/html/document.zig (hunk 3 above)
Comment (krichprollsch) [reply src/html/document.zig]: oh good point, indeed, I can just cast the document :facepalm: 
I'm updating

Comment (krichprollsch) [reply src/html/document.zig]: I'm also applying the same change in `dom/document.zig` when possible.

**Pull request merged**